sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1825,6 +1825,33 @@ class Dataset[T] private[sql](
     RelationalGroupedDataset(toDF(), cols.map(_.expr), RelationalGroupedDataset.CubeType)
   }
 
+  /**
+   * Create multi-dimensional aggregation for the current Dataset using the specified grouping sets,
+   * so we can run aggregation on them.
+   * See [[RelationalGroupedDataset]] for all the available aggregate functions.
+   *
+   * {{{
+   *   // Compute the average for all numeric columns group by specific grouping sets.
+   *   ds.groupingSets(Seq(Seq($"department", $"group"),Seq()),$"department", $"group").avg()
```

Comment:
The SQL syntax is `GROUP BY ... GROUPING SETS (...)`, shall we put grouping cols as the first parameter?

Reply:
This is because the * parameter must be placed at the end.

Reply:
oh I see, then no problem